### PR TITLE
make the constant MAX_LINE_SIZE configurable

### PR DESCRIPTION
**Pull request by sjnam**
local form, err = upload:new()
if not form then
   ngx.log(ngx.ERR, "failed to new upload: ", err)
   ngx.exit(500)
end

local max_line_size = 1024
form:set_line_size(max_line_size)

### CODE REVIEW

**Comment by agentzh** [comment]
Maybe we can just make the max_line_size the (optional) second argument to the `new` method?

**Comment by sjnam** [comment]
Is this OK?

**Comment by agentzh** [comment]
@sjnam Will you please add some test cases for this feature to the existing test suite? Otherwise it looks good to me :) Thanks a lot!

**Comment by sjnam** [comment]
I added test case "TEST 10" which is almost identical to "TEST 2" except the following line

local form = upload:new(5, 1024) -- max_line_size = 1024

so this test case has not an error. 
TEST 2 has an "fail to read: line too long" error.

**Comment by agentzh** [comment]
@sjnam That's cool. Thank you. I'll look into this PR soon.

File: t/sanity.t [hunk 1]
```diff
@@ -590,3 +590,59 @@ read: ["eof"]
 --- no_error_log
 [error]
 
+
```
Comment (agentzh) [inline t/sanity.t]: Style: we use 3 blank lines to separate test blocks (we usually use the [reindex](https://raw.githubusercontent.com/agentzh/old-openresty/master/bin/reindex) script to automatically format `.t` files for us).

**Comment by sjnam** [comment]
@agentzh Happy New Year! :-) I added one more blank line.

**Comment by agentzh** [comment]
I'll check this out soon, thank you. And happy new year :)

**Comment by agentzh** [comment]
@sjnam Sorry for the delay on my side. Just merged your changes to git master :) Thanks a lot!

**Pull request closed without merging**